### PR TITLE
Extract shared wind section for full-* embeds

### PR DESCRIPTION
**Pull request by alexwitherspoon**
## Description

`full-single`, `full-dual`, and `full-multi` each carried a byte-identical copy of the compass + wind-details markup (~70 lines each). Extract it into one `buildFullWindSection()` helper and call it from all three, removing ~210 lines of duplication. This sets up the wind-forward weather-region redesign to land in one shared place (follow-up), mirroring how `wind-visual.js` was extracted before enhancing.

## Type

- [x] Refactor

## Behavior change (intentional, minor)

The markup is otherwise identical, with one deliberate correctness fix surfaced during review: the "Peak Gust @ Time" row now requires `peak_gust_time > 0` (previously `!== null`). A zero/invalid timestamp (e.g. a failed `strtotime()` upstream) would have rendered the epoch (midnight); it now falls back to `--`. Output is unchanged for all valid timestamps.

## Testing

- `make test-ci` passes (0 errors).
- All three full-* styles render locally (HTTP 200); output is identical to before except the peak-time edge case noted above.

## Checklist

- [x] Tests pass
- [x] Documentation updated (if needed) - n/a
- [x] No sensitive data (API keys, passwords, credentials)
- [x] Code follows [CODE_STYLE.md](CODE_STYLE.md)

## Follow-up

Next PR: the wind-forward redesign of the full-* weather region (larger compass, Gust Factor + legend, METAR summary, compact metric columns) applied to `buildFullWindSection()`; webcam layout untouched.

## Related

Continues the embed wind-forward work (#161, #162, #163).

### CODE REVIEW

File: lib/embed-templates/full.php [hunk 1]
```diff
@@ -206,6 +206,98 @@ function buildFullWidgetMetrics($weather, $options, $hasMetarData) {
...
+ * @param string $windUnitLabel Wind speed unit label (e.g. "kt")
+ * @param float|null $windSpeed Wind speed in knots (for the Speed row)
+ * @param string $windUnit Wind speed unit code (kt/mph/kmh)
+ * @param float|null $gustSpeed Gust speed in knots
+ * @param float|null $peakGustToday Today's peak gust in knots
+ * @param int|null $peakGustTime Peak gust observation time (Unix seconds)
+ * @param string $timezone Airport timezone for the peak gust time
+ * @return string Wind section HTML
+ */
+function buildFullWindSection(string $canvasId, string $windDir, $windSpd, string $gustVal, string $windUnitLabel, $windSpeed, string $windUnit, $gustSpeed, $peakGustToday, $peakGustTime, string $timezone): string {
```
[the remark this answers is not in the document]
Comment (alexwitherspoon) [reply lib/embed-templates/full.php]: Fixed in 4873071: added nullable type hints for the numeric params (?float speeds, ?int peak gust time) and corrected the $windSpd doc to float|string (it's a round() result or '---'), so the signature and PHPDoc agree.

File: lib/embed-templates/full.php [hunk 2]
```diff
@@ -206,6 +206,98 @@ function buildFullWidgetMetrics($weather, $options, $hasMetarData) {
...
+        $html .= '--';
+    }
+    $html .= <<<HTML
+</span>
+                    </div>
+                    <div class="metric-item peak-time-item">
+                        <span class="label">@ Time</span>
+                        <span class="value">
+HTML;
+    if ($peakGustTime !== null && $peakGustToday !== null && $peakGustToday > 0) {
```
[the remark this answers is not in the document]
Comment (alexwitherspoon) [reply lib/embed-templates/full.php]: Fixed in a76b5e6: the "@ Time" row now requires peak_gust_time > 0, so a zero/invalid timestamp falls back to '--' instead of rendering the epoch.

File: lib/embed-templates/full.php [hunk 3]
```diff
@@ -206,6 +206,98 @@ function buildFullWidgetMetrics($weather, $options, $hasMetarData) {
...
+ * @param string $windUnitLabel Wind speed unit label (e.g. "kt")
+ * @param float|null $windSpeed Wind speed in knots (for the Speed row)
+ * @param string $windUnit Wind speed unit code (kt/mph/kmh)
+ * @param float|null $gustSpeed Gust speed in knots
+ * @param float|null $peakGustToday Today's peak gust in knots
+ * @param int|null $peakGustTime Peak gust observation time (Unix seconds)
+ * @param string $timezone Airport timezone for the peak gust time
+ * @return string Wind section HTML
+ */
+function buildFullWindSection(string $canvasId, string $windDir, $windSpd, string $gustVal, string $windUnitLabel, ?float $windSpeed, string $windUnit, ?float $gustSpeed, ?float $peakGustToday, ?int $peakGustTime, string $timezone): string {
```
[the remark this answers is not in the document]
Comment (alexwitherspoon) [reply lib/embed-templates/full.php]: Fixed in 128e082: $windSpd now has an explicit float|string type hint, so buildFullWindSection's signature is fully typed.

File: lib/embed-templates/full.php [hunk 4]
```diff
@@ -206,6 +206,98 @@ function buildFullWidgetMetrics($weather, $options, $hasMetarData) {
...
+                    </div>
+                    <div class="metric-item peak-time-item">
+                        <span class="label">@ Time</span>
+                        <span class="value">
+HTML;
+    if ($peakGustTime > 0 && $peakGustToday !== null && $peakGustToday > 0) {
+        try {
+            $tz = new DateTimeZone($timezone);
+            $dt = new DateTime('@' . $peakGustTime);
+            $dt->setTimezone($tz);
```
[the remark this answers is not in the document]
Comment (alexwitherspoon) [reply lib/embed-templates/full.php:283]: Good point - updated the PR description to call out that one intentional change: the "@ Time" row now requires peak_gust_time > 0 (was !== null), so a zero/invalid timestamp falls back to '--' instead of rendering the epoch. Everything else is byte-identical.
